### PR TITLE
add python h5py package

### PR DESCRIPTION
**Pull request by rikenm1**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Golden container needs python-h5py python package. 

###### Change Log  

- Change : Move libaec, liblzf, time to core.
- Change: Added hdf5, hostname, python-cached-property, python-pkgconfig and python-h5py to core.

###### Does this affect the toolchain?  

**YES/NO**
No

###### Test Methodology

- Pipeline build id: 264707

### CODE REVIEW

File: SPECS/hdf5/hdf5.spec [hunk 1]
```diff
@@ -0,0 +1,1054 @@
+%global macrosdir %(d=%{_rpmconfigdir}/macros.d; [ -d $d ] || d=%{_sysconfdir}/rpm; echo $d)
+# No more Java on i686
+%ifarch %{java_arches}
+%bcond_without java
+%else
+%bcond_with java
+%endif
```
Comment (PawelWMS) [inline SPECS/hdf5/hdf5.spec:7]: We always have Java, so we can drop these `bcond_with*` macros and keep the parts of the spec where `%{with java}` evaluates to `true`.

File: SPECS/hdf5/hdf5.spec [hunk 2]
```diff
@@ -0,0 +1,1054 @@
...
+%global version_main %(echo %{version} | cut -d. -f-2)
```
Comment (PawelWMS) [inline SPECS/hdf5/hdf5.spec:8]: I think this should fix the CGManifest check.

```suggestion
%define version_main %(echo %{version} | cut -d. -f-2)
```

File: SPECS/hdf5/hdf5.spec [hunk 3]
```diff
@@ -0,0 +1,1054 @@
...
+%global with_openmpi 0
+%if %{with_mpich}
+%global mpi_list mpich
+%endif
+%if %{with_openmpi}
+%global mpi_list %{?mpi_list} openmpi
+%endif
+Summary:        A general purpose library and file format for storing scientific data
+# Patch version?
+#global snaprel -beta
```
Comment (PawelWMS) [inline SPECS/hdf5/hdf5.spec:20]: Please remove unused/unhelpful comments.
```suggestion
```

File: SPECS/hdf5/hdf5.spec [hunk 4]
```diff
@@ -0,0 +1,1054 @@
+%global macrosdir %(d=%{_rpmconfigdir}/macros.d; [ -d $d ] || d=%{_sysconfdir}/rpm; echo $d)
+# No more Java on i686
+%ifarch %{java_arches}
```
Comment (PawelWMS) [inline SPECS/hdf5/hdf5.spec:3]: We don't have `java_arches` defined in our build environment, so we shouldn't use it (just assume Java is there - see my comment below). I'm guessing this is some Fedora-specific macro,

File: SPECS/hdf5/hdf5.spec [hunk 5]
```diff
@@ -0,0 +1,1054 @@
...
+BuildRequires: time
+BuildRequires: zlib-devel
+BuildRequires: hostname
+# For patches/rpath
+BuildRequires: automake
+BuildRequires: libtool
+# Needed for mpi tests
+BuildRequires: openssh-clients
+BuildRequires: libaec-devel
+BuildRequires: gcc, gcc-c++
```
Comment (PawelWMS) [inline SPECS/hdf5/hdf5.spec:73]: We do one package per line and keep them in alphabetical order. You can lint the spec and that should do it for you.

File: SPECS/hdf5/hdf5.spec [hunk 6]
```diff
@@ -0,0 +1,1054 @@
...
+Summary: HDF5 openmpi static libraries
+Requires: %{name}-openmpi-devel%{?_isa} = %{version}-%{release}
+
+%description openmpi-static
+HDF5 parallel openmpi static libraries
+%endif
+
+
+%prep
+%autosetup -n %{name}-%{version}%{?snaprel} -p1
```
Comment (PawelWMS) [inline SPECS/hdf5/hdf5.spec:179]: Since we don't have `snaprel`, we can simplify:
```suggestion
%autosetup -p1
```

File: SPECS/hdf5/hdf5.spec [hunk 7]
```diff
@@ -0,0 +1,1054 @@
...
+# Force shared by default for compiler wrappers (bug #1266645)
+sed -i -e '/^STATIC_AVAILABLE=/s/=.*/=no/' */*/h5[cf]*.in
+autoreconf -f -i
+
+# Modify low optimization level for gnu compilers
+sed -e 's|-O -finline-functions|-O3 -finline-functions|g' -i config/gnu-flags
+
+%build
+#Do out of tree builds
+%global _configure ../configure
```
Comment (PawelWMS) [inline SPECS/hdf5/hdf5.spec:207]: Not used.
```suggestion
```

File: SPECS/hdf5/hdf5.spec [hunk 8]
```diff
@@ -0,0 +1,1054 @@
...
+%endif
+
+%check
+%ifarch %{ix86} s390x
+# i386: java TestH5Arw segfaults
+# s390x: Testing inserting objects to create first direct block in recursive indirect blocks five levels deep*FAILED*
+make -C build check || :
+%else
+make -C build check
+%endif
```
Comment (PawelWMS) [inline SPECS/hdf5/hdf5.spec:338]: We don't build for these architectures, let's remove this part of the spec.
```suggestion
make -C build check
```

File: SPECS/hdf5/hdf5.spec [hunk 9]
```diff
@@ -0,0 +1,1054 @@
...
+#export HDF5_Make_Ignore=yes
+export OMPI_MCA_rmaps_base_oversubscribe=1
+# openmpi 5+
+export PRTE_MCA_rmaps_default_mapping_policy=:oversubscribe
+for mpi in %{?mpi_list}
+do
+  module load mpi/$mpi-%{_arch}
+# i686 & s390x mpich - testphdf5: malloc.c:4189: _int_malloc: Assertion `(unsigned long) (size) >= (unsigned long) (nb)' failed.
+%ifarch armv7hl %{ix86} s390x
```
Comment (PawelWMS) [inline SPECS/hdf5/hdf5.spec:347]: Similar comment about architectures as above.

File: SPECS/hdf5/hdf5.spec [hunk 10]
```diff
@@ -0,0 +1,1054 @@
...
+%else
+  make -C $mpi check
+%endif
+  module purge
+done
+
+# I have no idea why those get installed. But it's easier to just
+# delete them, than to fight with the byzantine build system.
+# And yes, it's using /usr/lib not %_libdir.
+if [ %_libdir != /usr/lib ]; then
```
Comment (PawelWMS) [inline SPECS/hdf5/hdf5.spec:358]: In our case `%_libdir` evaluates to `/usr/lib`, so we can remove this part and the comment.

File: SPECS/hostname/hostname.spec [hunk 11]
```diff
@@ -0,0 +1,234 @@
...
+if [ $1 -eq 0 ]; then
+  # Package removal, not upgrade...
+  systemctl --no-reload disable --now nis-domainname.service &>/dev/null || :
+fi
+
+# NOTE: Nothing to do for upgrade (in postun), nis-domainname.service is oneshot.
+
+%files
+%doc COPYRIGHT
+%{!?_licensedir:%global license %%doc}
```
Comment (PawelWMS) [inline SPECS/hostname/hostname.spec:56]:
```suggestion
```

File: SPECS/libaec/libaec.spec [hunk 12]
```diff
@@ -1,15 +1,14 @@
+Summary:        Adaptive Entropy Coding library
 Name:           libaec
 Version:        1.0.4
-Release:        4%{?dist}
-Summary:        Adaptive Entropy Coding library
-License:        BSD
+Release:        5%{?dist}
+License:        BSD 2-Clause
```
Comment (PawelWMS) [inline SPECS/libaec/libaec.spec:5]: Thanks for switching to [SPDX identifiers](https://spdx.org/licenses/)! I think we only missed a hyphen here.
```suggestion
License:        BSD-2-Clause
```

File: SPECS/liblzf/liblzf.spec [hunk 13]
```diff
@@ -70,6 +69,9 @@ rm -f %{buildroot}%{_libdir}/liblzf.la
 %{_libdir}/pkgconfig/liblzf.pc
 
 %changelog
+* Tue Nov 01 2022 Riken Maharjan <rmaharjan@microsoft.com> - 3.6-24
```
Comment (PawelWMS) [inline SPECS/liblzf/liblzf.spec:72]: Let's also remove the comment block starting with "Leave lzcat  out since it conflicts with xz-lzma". It looks like the command it's referring to is commented out as well.

File: SPECS/python-cached_property/python-cached_property.spec [hunk 14]
```diff
@@ -0,0 +1,184 @@
...
+%autosetup -p1 -n %{projectname}-%{version}
+
+%build
+%{__python3} setup.py build
+
+%install
+%{__python3} setup.py install -O1 --skip-build --root %{buildroot}
+
+%check
+pip3 install pytest freezegun
```
Comment (PawelWMS) [inline SPECS/python-cached_property/python-cached_property.spec:49]: We've had issues with `pytest` updates breaking tests in the past. Please take a look at the test output from the logs and freeze the version of `pytest` to whatever was used now.

File: SPECS/python-h5py/python-h5py.spec [hunk 15]
```diff
@@ -0,0 +1,342 @@
...
+cd serial
+%py3_install
+rm -rf %{buildroot}%{python3_sitearch}/h5py/tests
+cd -
+
+
+
+%files -n python%{python3_pkgversion}-h5py
+%license serial/licenses/*.txt
+#doc serial/ANN.rst serial/README.rst serial/examples
```
Comment (PawelWMS) [inline SPECS/python-h5py/python-h5py.spec:89]:
```suggestion
```

File: SPECS/time/time.spec [hunk 16]
```diff
@@ -38,25 +38,25 @@ Release:    9%{?dist}
 # m4/stdnoreturn.m4:        FSFULLR
 # maint.mk:                 GPLv3+
 # tests/time-posix-quiet.sh:    GPLv3+
-License:    GPLv3+ and GFDL
+License:        GPLv3+ AND GFDL
```
Comment (PawelWMS) [inline SPECS/time/time.spec:41]: The comments above suggest a couple more licenses. Could you double-check, please and either update the `License` or the comments?

**Review by PawelWMS — CHANGES_REQUESTED** [review]
Some comments in the files and two more things outside the PR:
- From the builds it looks like `h5py` and `python-cached_property` tests are failing - we'd need to fix that before we merge.
- Did you also run a buddy build for ARM64?

File: SPECS/hdf5/hdf5.spec [hunk 17]
```diff
@@ -0,0 +1,971 @@
...
+%{_libdir}/openmpi/share/man/man1/h5pfc.1*
+
+%files openmpi-static
+%{_libdir}/openmpi/lib/*.a
+%endif
+
+
+%changelog
+* Tue Nov 01 2022 Riken Maharjan <rmaharjan@microsoft.com> - 1.12.1-11
+- License verified
```
Comment (PawelWMS) [inline SPECS/hdf5/hdf5.spec:434]: Please also mention we dropped the Java subpackages with this change.

File: SPECS/hdf5/hdf5.spec [hunk 18]
```diff
@@ -0,0 +1,971 @@
...
+BuildRequires: zlib-devel
+BuildRequires: hostname
+# For patches/rpath
+BuildRequires: automake
+BuildRequires: libtool
+# Needed for mpi tests
+BuildRequires: gcc
+BuildRequires: gcc-c++
+BuildRequires: libaec-devel
+BuildRequires: openssh-clients
```
Comment (PawelWMS) [inline SPECS/hdf5/hdf5.spec:56]: Please lint the spec - it looks like we have some duplicates and inconsistent spacing.

Comment (rikenm1) [reply SPECS/hdf5/hdf5.spec:56]: spec linting was crashing when I used the linter with this file. let me try again with all the changes I applied. Except for this hdf5 spec, I lint'd all other spec files.

Comment (rikenm1) [reply SPECS/hdf5/hdf5.spec]: @PawelWMS linting works now. I should have saved what error I was getting yesterday.

File: SPECS-EXTENDED/slf4j/LICENSE-2.0.txt [hunk 19]
```diff
@@ -1,202 +1,202 @@
-
```
Comment (PawelWMS) [inline SPECS-EXTENDED/slf4j/LICENSE-2.0.txt:1]: I have no idea what is the diff that git sees (there seems to be no actual changes), but let's fix this to not overwrite the file's history. I think something along the lines of of `git checkout main SPECS-EXTENDED/slf4j` should do the trick.

File: SPECS/hdf5/hdf5.spec (hunk 18 above)
Comment (PawelWMS) [reply SPECS/hdf5/hdf5.spec]: No worries, we can take a look, if we hit it again.

File: SPECS-EXTENDED/slf4j/LICENSE-2.0.txt (hunk 19 above)
Comment (rikenm1) [reply SPECS-EXTENDED/slf4j/LICENSE-2.0.txt]: SPECS-EXTENDED/slf4j does the trick

**Review by PawelWMS — APPROVED** [review]
